### PR TITLE
Update dependencies commons-beanutils and bouncycastle-api

### PR DESCRIPTION
**Pull request by alxsap**
### Proposed changelog entries

- Upgrade of commons-beanutils to [1.9.4](http://commons.apache.org/proper/commons-beanutils/changes-report.html#a1.9.4)
- Upgrade of bouncycastle-api to [2.18](https://github.com/jenkinsci/bouncycastle-api-plugin/releases/tag/bouncycastle-api-2.18)

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [N/A] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [N/A] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by MRamonLeon — COMMENTED** [review]
A lot of test faiures with:

```
master50406|Caused by: org.apache.commons.jelly.JellyException: jar:file:/home/jenkins/workspace/Core_jenkins_PR-4928/ath/categoriesfirefoxjdk8/target/jenkins2971151567829325209home/war/WEB-INF/lib/jenkins-core-2.257-SNAPSHOT.jar!/jenkins/model/OptionalJobProperty/config.jelly:30:97: <st:include> This tag does not understand the 'class' attribute
```

**Comment by jvz** [comment]
That's from commons-beanutils. I remember we tried to upgrade this dependency back when it was originally released and saw the same test failures.

**Comment by varyvol** [comment]
@alxsap did you have the opportunity to review the test failures and the reasons for it? Thanks!

**Comment by alxsap** [comment]
@varyvol Not yet, sorry

**Comment by daniel-beck** [comment]
We rely on the "bad" behavior in commons-beanutils. This won't work, I suggest you don't bother unless you're prepared to spend a lot of time making things work.

File: bom/pom.xml [hunk 1]
```diff
@@ -241,7 +241,7 @@ THE SOFTWARE.
       <dependency>
         <groupId>commons-beanutils</groupId>
         <artifactId>commons-beanutils</artifactId>
-        <version>1.9.3</version>
+        <version>1.9.4</version>
```
Comment (oleg-nenashev) [inline bom/pom.xml:244]:
```suggestion
        <version>1.9.3</version>
```

**Review by oleg-nenashev — COMMENTED** [review]
What are the next steps here. We will eventually need to update commons-beanutils, and I agree with @daniel-beck that this is not trivial. Is there an interest to proceed with this pull request? Or should we close it / exclude beanutils?

File: bom/pom.xml (hunk 1 above)
Comment (dbreheret) [reply bom/pom.xml:244]: I am interested by the PR, our security AppScan are not happy about this commons-beanutils 3rd party.

https://vuln.whitesourcesoftware.com/vulnerability/CVE-2014-0114
which is fixed on version 1.9.4

Comment (jvz) [reply bom/pom.xml:244]: If you can help fix the failing tests caused by the upgrade, we can upgrade.

Comment (daniel-beck) [reply bom/pom.xml:244]: And that was also caused by an upgrade of the library. We know doing that breaks things, that's the whole problem.

**Comment by daniel-beck** [comment]
This PR doesn't work as is, and there was no response to a previous comment two weeks ago, so I'm closing this as housekeeping.

An update of commons-beanutils would be welcome, but it needs to work.

**Comment by dbreheret** [comment]
@daniel-beck 
I did a quick investigation.
I think it is due to this security fix : CVE-2019-10086. Apache Commons Beanutils does not suppresses the class property in bean introspection by default.
See detail on https://commons.apache.org/proper/commons-beanutils/

I have tried with on the following failing test BuildDiscarderPropertyTest.buildDiscarderField().

If I call 
```
PropertyUtilsBean pub = BeanUtilsBean.getInstance().getPropertyUtils();
pub.removeBeanIntrospector(SuppressPropertiesBeanIntrospector.SUPPRESS_CLASS);
```
the test becomes green.

I think we need someone that knows Jelly well (unlike me) to investigate the "class" attribute, like in 
"config.jelly:30:97: st:include This tag does not understand the 'class' attribute".

**Comment by daniel-beck** [comment]
@dbreheret

> I think we need someone that knows Jelly well (unlike me) to investigate the "class" attribute, like in

Agreed, which is why this PR as is isn't feasible, and why I closed it.

**Pull request closed without merging**